### PR TITLE
Fix leader election with RBAC in tests

### PR DESCRIPTION
**Pull request by munnerz**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by munnerz** [comment]
This PR also disables scoping Navigator to a single namespace as part of the standard helm chart deployment

File: contrib/charts/navigator/templates/rbac.yaml [hunk 1]
```diff
@@ -95,6 +95,10 @@ items:
   - apiGroups: ["apps"]
     resources: ["statefulsets"]
     verbs:     ["get", "list", "watch", "update", "create", "delete"]
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs:     ["*"]
+    resourceNames: ["navigator-controller"]
```
Comment (wallrj) [inline contrib/charts/navigator/templates/rbac.yaml]: ❓ What does this do?

**Comment by wallrj** [comment]
[log.gz](https://github.com/jetstack/navigator/files/1430669/log.gz)

**Comment by munnerz** [comment]
So after looking into this some more, the actual controller won't be able to create the initial Endpoint resource used for leader election as we have explicitly named the resource in the RBAC policy, which consequently then rejects POST requests to the API endpoint (as the endpoint does *not* include the resource name at this point). This is a limitation with RBAC at the moment.

File: contrib/charts/navigator/templates/rbac.yaml (hunk 1 above)
Comment (munnerz) [reply contrib/charts/navigator/templates/rbac.yaml]: This allows navigator-controller to modify Endpoints resources with the name `navigator-controller`. This is the resource used for leader election 😄

**Comment by munnerz** [comment]
Updated & tested working. Review as and when you've got time. Don't want to hold you up!

**Comment by munnerz** [comment]
@wallrj - I'm well aware the e2e tests need improving. Currently we are blocked by needing to supply some e2e testing infrastructure via Prow, so that we can actually get VMs big enough to run Elasticsearch (it is a massive resource hog).

This will I'll be working on some e2e testing, so will get the actual test cases written up then so we can start to see how they perform 😄

**Pull request merged**